### PR TITLE
Documented the pe::header::DosHeader.

### PR DESCRIPTION
**Pull request by JohnScience**
I've noticed that there was little to no documentation of the DOS header, so I decided to add it. I've provided sources for all the claims that are made in the docs.

### CODE REVIEW

**Comment by JohnScience** [comment]
@m4b

**Review by m4b — APPROVED** [review]
this is really impressive documentation effort; thank you for putting the time, effort, and patience into writing this up and collecting all this information, along with links, great stuff, thank you!

**Comment by m4b** [comment]
I have to ask, was the documentation generated using some form of automation or with an AI assistant?

**Comment by m4b** [comment]
release note: non-breaking

File: src/pe/header.rs [hunk 1]
```diff
@@ -5,50 +5,288 @@ use alloc::vec::Vec;
...
-    /// e_cp
+    /// In `winnt.h` and `pe.h`, it's `e_cp`.
+    ///
+    /// It used to specify the number of pages required to hold a file. For example,
+    /// if the file contained 1024 bytes, and the file had pages of a size of 512 bytes,
+    /// this [word](https://en.wikipedia.org/wiki/Word_(computer_architecture)) would contain
+    /// 0x0002 (2 pages); if the file contained 1025 bytes, this word would contain 0x0003 (3 pages).
+    /// [Source](https://stixproject.github.io/data-model/1.2/WinExecutableFileObj/DOSHeaderType/).
+    ///
+    /// Typically, this field is set to 0. [Source](https://offwhitesecurity.dev/malware-development/portable-executable-pe/dos-header/).
```
[the remark this answers is not in the document]
Comment (JohnScience) [reply src/pe/header.rs:71]: It just happened so that this was a website that told about such details as typical values for the fields.

**Comment by JohnScience** [comment]
> I have to ask, was the documentation generated using some form of automation or with an AI assistant?

For the most part, the documentation was written manually with the assistance of GitHub Copilot. There was a small exception, though. I haven't found a website with decent explanation of one topic of Program Segment Prefix (PSP) so I fact-checked the paragraph generated by ChatGPT and used it to make the part about PSP easier to understand.

**Pull request merged**